### DISCUSSION TITLE
Blazor web assembly components in cms output templates

### DISCUSSION THREAD

**Discussion by kevinasdzine** [OrchardCMS/OrchardCore#16928] [Q&A]
Has anyone integrated Blazor WASM into an Orchard Core solution where a Blazor component was introduced to the output template of an orchard content type? I'd like to create a content type that presents instances using standard orchard shape rendering templates but those templates include Blazor components for some client side sugar.

**Comment by kevinasdzine** [comment]
I got it partially working but Orchard Core's behavior of lower-casing all element names is getting in the way of passing fragments to the Blazor component. Is there a way to selectively disable that behavior?

**Comment by ns8482e** [comment]
You can definitely use Blazor Web assembly  components in CMS output. Create your web assembly component library and you can add as project reference to your CMS project.

**Reply by kevinasdzine** [reply]
What wasn't particularly clear (although your post references it) was the requirement for the Blazor components to be in a separate project from the theme where they are being used. I was not successful in getting both a client-side wasm project and a server-side project to coexist in the same orchard core solution (I imagine .NET 9 will solve that).

I was pleasantly surprised that I could pass my ContentItem.Content JSON as a variable to a component instead of a string (I was concerned about it passing as serialized JSON and dealing with quote conflicts) which allowed me to pass a complex type with nested Bag parts (recursive parts, at that) and render that as a complicated client-side interactive component!
